scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -54,74 +53,54 @@ def load_light_scenarios_from_jsonl(path: str) -> List[LightScenario]:
     light_scenario_jsons = open(path, "r").readlines()
     for light_scenario_json in light_scenario_jsons:
         light_scenario_dict: dict = json.loads(light_scenario_json)
 
-        light_scenario_metadata: dict = light_scenario_dict["light_scenario_key"]["metadata"]
+        light_scenario_key_dict: dict = light_scenario_dict["scenario_key"]
         # if the light_scenarios are exported from helm, they will have a scenario_spec field
-        if "scenario_spec" in light_scenario_metadata:
-            light_scenario_metadata["scenario_spec"] = ScenarioSpec(**light_scenario_metadata["scenario_spec"])
-        light_scenario_key = LightScenarioKey(metadata=light_scenario_metadata)
+        scenario_spec = ScenarioSpec(**light_scenario_key_dict["scenario_spec"])
+        light_scenario_key = LightScenarioKey(scenario_spec=scenario_spec, split=light_scenario_key_dict["split"])
         light_instances: List[LightInstance] = [
-            create_light_instance_from_dict(instance_dict) for instance_dict in light_scenario_dict["light_instances"]
+            create_light_instance_from_dict(instance_dict) for instance_dict in light_scenario_dict["instances"]
         ]
-        light_scenarios.append(LightScenario(light_scenario_key=light_scenario_key, light_instances=light_instances))
+        light_scenarios.append(LightScenario(scenario_key=light_scenario_key, instances=light_instances))
     return light_scenarios
 
 
 def create_ngram_index(
-    light_scenarios: List[LightScenario], n_values: List[int], tokenizer: LightTokenizer
+    light_scenarios: List[LightScenario],
+    n_values: List[int],
+    tokenizer: LightTokenizer,
+    stats_key_counts: Dict[DataOverlapStatsKey, int],
 ) -> NgramIndex:
     """Given a list of scenarios and n values, initialize ngram_index"""
     ngram_index: NgramIndex = {n: {} for n in n_values}
     for scenario in light_scenarios:
-        hlog(f"Building ngram indexes for {scenario.light_scenario_key}")
+        hlog(f"Building ngram indexes for {scenario.scenario_key}")
         for n in n_values:
-            stats_key = DataOverlapStatsKey(metadata={"light_scenario_key": scenario.light_scenario_key, "N": n})
-            for i in range(len(scenario.light_instances)):
-                instance = scenario.light_instances[i]
+            stats_key = DataOverlapStatsKey(
+                light_scenario_key=scenario.scenario_key, overlap_protocol_spec=OverlapProtocolSpec(n=n)
+            )
+            num_instances = len(scenario.instances)
+            stats_key_counts[stats_key] = num_instances
+            for i in range(num_instances):
+                instance = scenario.instances[i]
                 input_tokens = tokenizer.tokenize(instance.input)
                 for input_ngram in ngrams(input_tokens, n):
                     if input_ngram not in ngram_index[n]:
                         ngram_index[n][input_ngram] = set()
+                    id = instance.id if instance.id else f"id{i}"
```

Comment:
Is there actually a scenario where instance.id is empty? This seems risky in that you could get ID collisions.

Reply:
Let's just do an assert here (which will make the typechecker happy).

Reply:
Sounds good, updated to assert